opengrok-indexer/src/test/java/org/opengrok/indexer/history/GitRepositoryTest.java

```diff
@@ -350,47 +337,44 @@ public void testGetHistoryForTwiceRenamed() throws Exception {
                 + "\treturn 0;\n"
                 + "}\n";
 
-        runRenamedTest(Paths.get("moved", "renamed.c").toString(), "1086eaf", exp_str);
-        runRenamedTest(Paths.get("moved", "renamed2.c").toString(), "67dfbe2", exp_str);
-    }
+        // old content (after revision 1086eaf5 inclusively)
+        runRenamedTest(Paths.get("moved2", "renamed2.c").toString(), "84599b3c", new_content);
+        runRenamedTest(Paths.get("moved2", "renamed2.c").toString(), "67dfbe26", new_content);
+        runRenamedTest(Paths.get("moved2", "renamed2.c").toString(), "1086eaf5", new_content);
 
-    /**
-     * Test that {@code getHistoryGet()} returns historical contents of renamed
-     * file.
-     *
-     * @throws java.lang.Exception
-     */
-    @Test
-    public void testGetHistoryForThreeTimesRenamed() throws Exception {
-        String exp_str
-                = "#include <stdio.h>\n"
-                + "#include <stdlib.h>\n"
-                + "\n"
-                + "int main ( int argc, const char * argv[] )\n"
-                + "{\n"
-                + "\tint i;\n"
-                + "\tfor ( i = 1; i < argc; i ++ )\n"
-                + "\t{\n"
-                + "\t\tprintf ( \"%s called with %d\\n\", argv [ 0 ], argv [ i ] );\n"
-                + "\t}\n"
-                + "\n"
-                + "\treturn 0;\n"
-                + "}\n";
+        runRenamedTest(Paths.get("moved", "renamed2.c").toString(), "67dfbe26", new_content);
+        runRenamedTest(Paths.get("moved", "renamed2.c").toString(), "1086eaf5", new_content);
+
+        runRenamedTest(Paths.get("moved", "renamed.c").toString(), "1086eaf5", new_content);
 
-        runRenamedTest(Paths.get("moved", "renamed.c").toString(), "b641394", exp_str);
-        runRenamedTest(Paths.get("renamed.c").toString(), "ce4c98e", exp_str);
+        // old content (before revision b6413947 inclusively)
+        runRenamedTest(Paths.get("moved2", "renamed2.c").toString(), "b6413947", old_content);
```

Comment:
these look like a candidate for array + forEach() ?

Reply:
Is it more readable though? I would leave it this way so a test failure can point to a particular line and also it's not that much lines.

Reply:
If you complement the assert with a message it should be enough.

Reply:
done